packages/gatsby-source-directus/gatsby-node.js

```diff
@@ -125,28 +125,28 @@ class Plugin {
 		this.urlGraphql = '';
 		this.url = '';
 		this.refreshInterval = 0;
+		this.authPromise = null;
 	}
 
 	async setOptions(options) {
 		const { url, dev, auth } = options;
 
 		if (isEmpty(url)) error('"url" must be defined');
 
+		if (this.directus) return this.authPromise;
+
 		const hasAuth = !!auth;
 		const hasToken = !isEmpty(auth?.token);
 		const hasEmail = !isEmpty(auth?.email);
 		const hasPassword = !isEmpty(auth?.password);
 		const hasCredentials = hasEmail && hasPassword;
 
-		if (hasAuth && !hasToken && !hasCredentials)
-			error('"auth.token" or ("auth.email" and "auth.password") must be defined');
-
-		if (hasAuth && !hasToken) error('("auth.email" and "auth.password") must be defined if "auth.token" is not set');
+		if (hasAuth) {
+			if (!hasToken && !hasCredentials) error('"auth.token" or ("auth.email" and "auth.password") must be defined');
 
-		if (hasToken && hasCredentials)
-			warning('"auth.token", "auth.email" and "auth.password" are all set, but only "auth.token" will be used');
-
-		if (!hasAuth) warning('no "auth" option were defined. Resources will be fetched with public role');
+			if (!hasToken && (!hasEmail || !hasPassword))
```

Comment:
Not sure am I seeing this correctly, but I believe technically `(!hasEmail || !hasPassword)` is the same as `!(hasEmail && hasPassword)` which ended up being `!hasCredentials`, so now it's the same condition as line 145 and feel slightly odd 🤔

On a side note, this "either token or credentials" case was also one of the things I had to think of when I was doing the 3rd party integrations so it'll definitely be helpful 😄

Reply:
Yeah, it makes total sense.
I have removed the second condition.
Although, in Gatsby v4 we have `pluginOptionsSchema` which allows to pass a Joi schema. Then the options will be tested against that schema, so I believe this conditions are not required at all. But just to be sure and since it makes the code more readable I think we should preserve them. 👍